lib/template-loader.php

```diff
@@ -105,8 +105,56 @@ function gutenberg_find_template( $template_file ) {
 	);
 
 	if ( $template_query->have_posts() ) {
-		$template_posts            = $template_query->get_posts();
-		$_wp_current_template_post = array_shift( $template_posts );
+		$template_posts        = $template_query->get_posts();
+		$current_template_post = array_shift( $template_posts );
+
+		// Build map of template slugs to their priority in the current hierarchy.
+		$slug_priorities = array_flip( $slugs );
+
+		// See if there is a theme block template with higher priority than the resolved template post.
+		$higher_priority_block_template_path     = null;
+		$higher_priority_block_template_priority = PHP_INT_MAX;
+		$block_template_files                    = glob( get_stylesheet_directory() . '/block-templates/*.html', 1 );
```

Comment:
do we really need this to be in a separate folder? Can't we use the root folder? @mtias may have thoughts here as well.

Reply:
It makes it easier to resolve files and avoids potential future naming conflicts.